core/shared/src/main/scala/fs2/pipe.scala

```diff
@@ -135,6 +136,28 @@ object pipe {
     }
   }
 
+  /** Debounce the stream with a minimum delay of `d` between each element */
+  def debounce[F[_], I](d: FiniteDuration)(implicit F: Effect[F], scheduler: Scheduler, ec: ExecutionContext): Pipe[F, I, I] = {
```

Comment:
What do you think about implementing this with `in => pipe2.mergeHaltR(time.sleep[F](d), in).scan(...)`?